lib/honeydew/worker.ex

```diff
@@ -14,9 +14,16 @@ defmodule Honeydew.Worker do
     defstruct [:queue, :queue_pid, :module, :user_state]
   end
 
+  def child_spec(opts) do
+    %{
+      id: __MODULE__,
+      start: {__MODULE__, :start_link, opts}
+    }
+  end
+
   @doc false
-  def start_link(queue, %{ma: {module, args}, init_retry: init_retry_secs}, queue_pid) do
-    GenServer.start_link(__MODULE__, [queue, queue_pid, module, args, init_retry_secs])
+  def start_link(queue, %{ma: {module, args}, init_retry: init_retry_secs, restart: restart, shutdown: shutdown}, queue_pid) do
```

Comment:
hm, how come you're passing the supervisor's restart and shutdown settings down to the worker's genserver?

Reply:
Hmmm. The issue with DynamicSupervisor is there doesn't seem to be a good way to copy the restart and shutdown settings across children like you could with `simple_one_for_one`. I was hoping this might alleviate that problem, but I guess it would be better for these settings to still be contained by the supervisor. Perhaps we start an agent to hold those settings for use in `start_child`?